ce/amf/demands.ts

```diff
@@ -68,15 +69,25 @@ export class Demands extends EntityMap<YAMLDictionary, DemandBlock> {
 }
 
 export class DemandBlock extends Entity {
-
+  #environment: Record<string, string | number | boolean | undefined> = {};
```

Comment:
Not asking to change this, asking to understand:

Why have private fields with setters instead of just making the fields public?

Reply:
I could have made a write-only property, but I wanted to make *sure* nobody tried to read the private data.

Reply:
Oh, to stop readers. Duh Bill